chirp/drivers/baofeng_uv17.py

```diff
@@ -0,0 +1,747 @@
+    data += bytes(radio.MODEL, 'ascii')
+    return data
+
+
+def _upload(radio):
+    """Upload procedure"""
+    # Put radio in program mode and identify it
+    _do_ident(radio)
+
+    # Start communication
+    _start_communication(radio)
+
+    # Get memory map
+    memory_map = _get_memory_map(radio)
+
+    # UI progress
+    status = chirp_common.Status()
+    status.cur = 0
+    status.max = radio.WRITE_MEM_TOTAL // radio.BLOCK_SIZE
+    status.msg = "Cloning to radio..."
+    radio.status_fn(status)
+
+    for block_number in radio.BLOCK_ORDER:
+        # Choose a block number is memory map is corrupt
+        # This happens when te upload process is interrupted
+        if block_number not in memory_map:
+            memory_map[memory_map.index(165)] = block_number
+        block_index = memory_map.index(block_number) + 1
+        start_addr = block_index * 0x1000
+        data_start_addr = radio.BLOCK_ORDER.index(block_number) * 0x1000
+        for addr in range(start_addr, start_addr + 0x1000, radio.BLOCK_SIZE):
+
+            # sending the data
+            data_addr = data_start_addr + addr - start_addr
+            data = radio.get_mmap()[data_addr:data_addr + radio.BLOCK_SIZE]
+            frame = _make_frame(b"W", addr, radio.BLOCK_SIZE, data)
+            baofeng_common._rawsend(radio, frame)
+
+            # receiving the response
+            ack = baofeng_common._rawrecv(radio, 1)
+            if ack != b"\x06":
+                msg = "Bad ack writing block 0x%04x" % addr
+                raise errors.RadioError(msg)
+
+            # UI Update
+            status.cur = (data_addr) // radio.BLOCK_SIZE
+            status.msg = "Cloning to radio..."
+            radio.status_fn(status)
+
+
+@directory.register
+class UV17(baofeng_uv17Pro.UV17Pro):
+    """Baofeng UV-17"""
+    VENDOR = "Baofeng"
+    MODEL = "UV-17"
+    NEEDS_COMPAT_SERIAL = False
+
+    BLOCK_ORDER = [16, 17, 18, 19,  24, 25, 26, 4, 6]
+
+    MEM_TOTAL = 0x9000
+    WRITE_MEM_TOTAL = 0x9000
+    BLOCK_SIZE = 0x40
+    BAUD_RATE = 57600
+    _ident = b"PSEARCH"
+    _magics = [[b"PASSSTA", 3], [b"SYSINFO", 1],
+               [b"\x56\x00\x00\x0A\x0D", 13], [b"\x06", 1],
+               [b"\x56\x00\x10\x0A\x0D", 13], [b"\x06", 1],
+               [b"\x56\x00\x20\x0A\x0D", 13], [b"\x06", 1],
+               [b"\x56\x00\x00\x00\x0A", 11], [b"\x06", 1],
+               [b"\xFF\xFF\xFF\xFF\x0C\x55\x56\x31\x35\x39\x39\x39", 1],
+               [b"\02", 8], [b"\x06", 1]]
+    _fingerprint = b"\x06" + b"UV15999"
+
+    _tri_band = False
+    POWER_LEVELS = [chirp_common.PowerLevel("Low", watts=1.00),
+                    chirp_common.PowerLevel("High",  watts=5.00)]
+
+    LENGTH_NAME = 11
+    VALID_BANDS = [baofeng_uv17Pro.UV17Pro._vhf_range,
+                   baofeng_uv17Pro.UV17Pro._uhf_range]
+    SCODE_LIST = LIST_SCODE
+
+    MEM_FORMAT = """
+    #seekto 0x0030;
+    struct {
+      lbcd rxfreq[4];
+      lbcd txfreq[4];
+      u8 unused1:8;
```

Comment:
I didn't notice that you did this in the previous driver, but `u8 foo:8;` is not only redundant, but it makes the bitwise structure more complex (and thus slower) than it needs to be. You should only use bitfields when you need to access individual bits of a larger type. A byte where all eight bits are treated as one, should just be `u8 foo;`. We can queue that cleanup for the other driver, but it'd be good if we can fix that before we merge this one.

Reply:
Good one!! I am going to fix that.